demos/random_cut_demo/src/main/scala/io/hydrosphere/rrcf/impl/RRCF.scala

```diff
@@ -1,22 +1,36 @@
 package io.hydrosphere.rrcf.impl
 
+import java.io._
 
-class RRCF(val forestParams: ForestParams, initialData: Array[DataPoint]) {
 
-  require(initialData.length >= forestParams.treesNum * forestParams.samplesNum + forestParams.shingleSize - 1)
+class RRCF(val forestParams: ForestParams, val initialData: Array[DataPoint]) {
+
+  require(initialData.length == forestParams.warmupPointsNum)
 
   val trees = new Array[RRCT](forestParams.treesNum)
-  val shingledData: List[DataPoint] = initialData.sliding(forestParams.shingleSize, 1).map(x => DataPoint(x.flatMap(_.array).toArray)).toList
-  val shuffledInitShingles: List[DataPoint] = scala.util.Random.shuffle(shingledData)
+  val shingledData: Array[DataPoint] = initialData.sliding(forestParams.shingleSize, 1).map(x => DataPoint(x.flatMap(_.array))).toArray
+  assert(shingledData.length == forestParams.samplesNum)
+  //  val shuffledInitShingles: List[DataPoint] = scala.util.Random.shuffle(shingledData)

```

Comment:
do you plan on shuffling shingles, or is it just commented code?

Reply:
Yes, just an obsolete line